src/core/handle.ml

```diff
@@ -185,7 +199,69 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       (* Actually add the symbol to the signature. *)
       out 3 "(symb) %s ≔ %a\n" x.elt pp_term t;
       let d = if op then None else Some(t) in
-      let ss = Sig_state.add_symbol ss e Defin x a impl d in
+      (fst (Sig_state.add_symbol ss e Defin x a impl d), None)
+  | P_inductive(e, id, a, l)     ->
+      (* Add the inductive type in the signature *)
+      let (ss, sym_typ) = handle_symbol ss e Defin id [] a in
+      (* Add the constructors in the signature.  *)
+      let add_cons :
+        sig_state -> (ident * p_term) list-> sig_state * sym list =
+        fun ss l ->
+        let rec aux acc ss l =
+          match l with
+          | []   -> ss, acc
+          | (id, a)::q ->
+              let (ss, acc) = aux acc ss q in
+              let (ss, sym_cons) = handle_symbol ss e Const id [] a in
+              ss, sym_cons::acc
+        in
+        aux [] ss l
+      in
+      (*let add_cons :
+            sig_state * sym list -> ident * p_term -> sig_state * sym list
+        = fun(ss, cons_list) (id, a) ->
+        let (ss, sym_cons) = handle_symbol ss e Const id [] a in
+        (ss, sym_cons::cons_list)
+      in
+      let (ss, cons_list) = List.fold_left add_cons (ss, []) l in *)
```

Comment:
Why did you change this part of the code? The new definition is more complicated.

Reply:
It's just to keep the initial order of constructors.

Reply:
Before, this order was reversed.

Reply:
Ok but, in this case, I prefer to add after line 226: `let cons_list = List.rev cons_list in`.